sdk/src/prices/calculate-pool-prices.ts

```diff
@@ -71,38 +70,31 @@ function checkLiquidity(
 }
 
 type PoolObject = { pool: WhirlpoolData; address: PublicKey };
-function getMostLiquidPool(
-  mintA: Address,
-  mintB: Address,
+function getMostLiquidPools(
+  quoteTokenMint: PublicKey,
   poolMap: PoolMap,
-  config = defaultGetPricesConfig
-): PoolObject | null {
-  const { tickSpacings, programId, whirlpoolsConfig } = config;
-  const pools = tickSpacings
-    .map((tickSpacing) => {
-      const pda = PDAUtil.getWhirlpool(
-        programId,
-        whirlpoolsConfig,
-        AddressUtil.toPubKey(mintA),
-        AddressUtil.toPubKey(mintB),
-        tickSpacing
-      );
-
-      return { address: pda.publicKey, pool: poolMap[pda.publicKey.toBase58()] };
-    })
-    .filter(({ pool }) => pool != null);
+): Record<string, PoolObject> {
+  const mostLiquidPools = new Map<string, PoolObject>();
+  Object.entries(poolMap).forEach(([address, pool]) => {
+      const mintA = pool.tokenMintA.toBase58();
+      const mintB = pool.tokenMintB.toBase58();
+
+      if (pool.liquidity.isZero()) {
+          return;
+      }
+      if (!pool.tokenMintA.equals(quoteTokenMint) && !pool.tokenMintB.equals(quoteTokenMint)) {
+          return;
+      }
 
-  if (pools.length === 0) {
-    return null;
-  }
+      const baseTokenMint = pool.tokenMintA.equals(quoteTokenMint) ? mintB : mintA;
 
-  return pools.slice(1).reduce<PoolObject>((acc, { address, pool }) => {
-    if (pool.liquidity.lt(acc.pool.liquidity)) {
-      return acc;
-    }
+      const existingPool = mostLiquidPools.get(baseTokenMint);
+      if (!existingPool || pool.liquidity.gt(existingPool.pool.liquidity)) {
```

Comment:
i suppose we are not changing anything here, but is there an inherent bias towards high TS pools since on an equal price impact basis, there's technically more liquidity in each tick since each tick represents a larger bp?

Reply:
don't let this block this PR, but just want to confirm

Reply:
When there are pools with different tickSpacing for the same token pair, I think the pools with higher fee rate are disadvantageous for the ExactOut mode estimate.
This is because more of the input to output 1000 USDC would be consumed as fees, and as a result, more input tokens are required.

Where this selection logic becomes not-optimal is when there is significant liquidity remaining where the price has moved slightly in the trade direction from the current price.
That is, a pool where liquidity is highly concentrated at a single tick and just slightly outside of that range (e.g. stable pairs like USDT/USDC).
In this case, the liquidity field of the Whirlpool account will have a smaller value, but the liquidity will be larger than in the other pools as it moves in the trade direction, and the trade results will be better.